src/commons/achievement/utils/AchievementInferencer.ts

```diff
@@ -359,51 +381,72 @@ class AchievementInferencer {
     node.displayDeadline = descendantDeadlines.reduce(compareDeadlines, node.displayDeadline);
   }
 
+  private generateMaxExp(node: InferencerNode) {
```

Comment:
generateSumExp. name is misleading

Reply:
It is to generate the maxExp of an achievement. Naming wont be consistent if the function is called generateSumExp while we have getMaxExp and the attribute is named maxExp.